spec/requests/spree/checkout_controller_spec.rb

```diff
@@ -49,4 +50,30 @@
       expect(SolidusBolt::Users::SyncPaymentSourcesService).to have_received(:call)
     end
   end
+
+  describe 'PATCH /checkout/update/confirm' do
+    let(:order) { FactoryBot.create(:order_with_totals) }
+
+    before do
+      # use test preparation from solidusio/solidus/frontend/spec/controllers/spree/checkout_controller_spec.rb
+      # because Spree::TestingSupport::OrderWalkthrough.up_to(:confirm) doesn't work
+      order.update! user: user
+      order.update(state: 'confirm')
+      create(:payment, amount: order.total, order: order)
+      order.create_proposed_shipments
+      order.payments.reload
+
+      allow(SolidusBolt::Accounts::AddAddressService).to receive(:call)
+
+      patch '/checkout/update/confirm'
+    end
+
+    it 'redirects to completion route' do
+      expect(response).to redirect_to spree.order_path(order)
+    end
+
+    it 'calls the service to add addresses' do
+      expect(SolidusBolt::Accounts::AddAddressService).to have_received(:call).twice
```

Comment:
You missed checking that this service is called with the right parameters. 
(I see that the same happens in other places)

Reply:
fixed in https://github.com/nebulab/solidus_bolt/pull/96/commits/9b96b27b52d35163300716f53ad0c61187da3d7a